src/include/storage/index/index_defs.h

```diff
@@ -0,0 +1,108 @@
+#pragma once
+
+#include <vector>
+#include "catalog/catalog_defs.h"
+#include "type/type_id.h"
+
+namespace terrier::storage::index {
+
+/**
+ * A column of the index key has an identifier, type, and describes whether it can be null.
+ */
+class IndexKeyColumn {
+ public:
+  /**
+   * Non-varlen constructor for index key columns.
+   * @param oid key column oid
+   * @param nullable whether the column is nullable
+   * @param type_id the non-varlen type of the column
+   */
+  IndexKeyColumn(catalog::indexkeycol_oid_t oid, type::TypeId type_id, bool nullable) : oid_(oid), packed_type_(0) {
```

Comment:
This class seems a lot like Column in catalog::schema except that it uses indexkeycol_oid_t. 
I do not see how using col_oid_t would break things. Could you explain do we need indexkeycol_oid_t? I feel like the values should be the same.

Reply:
The col_oid_t and indexkeycol_oid_t represent different concepts, indexkeycol is more ephemeral than the columns of a table. There can also be indexkey columns that do not map directly to a column in a table, e.g. index on (2*A + B). So we think maybe separating them is clearer.

Though we agree the name is bad...

Reply:
I see. Thanks for explaining.